modules/apps/27-interchain-accounts/keeper/handshake.go

```diff
@@ -85,22 +91,33 @@ func (k Keeper) OnChanOpenTry(
 	}
 
 	// Register interchain account if it does not already exist
-	k.RegisterInterchainAccount(ctx, counterparty.PortId)
+	accAddr := strings.TrimPrefix(version, fmt.Sprintf("%s-", types.Version))
+	if err := k.RegisterInterchainAccount(ctx, accAddr, counterparty.PortId); err != nil {
+		return err
+	}
+
```

Comment:
We shouldn't use relayer input for generating the account address. We should derive the expected address ourself and use that (while checking it matches that set in the version string)

Reply:
exactly this :+1: